add tail inotify debug fix and stage 2.31.10

PettitWesley opened the pull request:
*Issue #, if available:*

fixes #639 #637 

*Description of changes:*


By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

PettitWesley committed:
cherry-picks: add tail inotify debug fix

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

PettitWesley commented:
Already fixed in upstream Fluent Bit: https://github.com/fluent/fluent-bit/commit/3e360f42feb2e5d9de45258afb20630177f61127

PettitWesley committed:
Bump to 2.31.10

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

matthewfala reviewed (approved):
Looks good! thank you.

PettitWesley committed:
init: support parsing metadata response cluster ARN to name

> Cluster
> The Amazon Resource Name (ARN) or short name of the Amazon ECS cluster to which the task belongs.

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task-metadata-endpoint-v4.html

Previously we assumed the response was always just the name.

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

PettitWesley committed:
Added changelog entry for cluster init parsing fix

Signed-off-by: Wesley Pettit <wppttt@amazon.com>

matthewfala commented on init/fluent_bit_init_process_test.go:
```diff
@@ -59,23 +60,45 @@ func TestGetECSTaskMetadata(t *testing.T) {
 		ECS_TASK_ARN:          "arn:aws:ecs:us-west-2:123456789123:task/ecs-cluster/4ca5a280e68947cd84a8357f0d008fb5",
 		ECS_TASK_ID:           "4ca5a280e68947cd84a8357f0d008fb5",
 		ECS_FAMILY:            "code_test_A",
-		ECS_LAUNCH_TYPE:       "", // empty lunch type
```
😄 🥪

matthewfala reviewed (approved):
New changes look good.
